api/src/main/java/org/openmrs/api/db/DiagnosisDAO.java

```diff
@@ -56,22 +57,27 @@
 	 * @throws DAOException exception thrown if error occurs while deleting the diagnosis
 	 */
 	void deleteDiagnosis(Diagnosis diagnosis) throws DAOException;
-	
+
 	/**
 	 * Gets all diagnoses for a given encounter
 	 *
-	 * @param encounter the specific encounter to get the diagnoses for.
-	 * @return list of diagnoses for an encounter
+	 * @deprecated since 2.5.0, use {@link #getDiagnosesByEncounter}
```

Comment:
We do not deprecate methods in the data access layer because we do not expect clients to access them directly. They should use the service layer, which then indirectly accesses the data access layer. So we just remove them.

Reply:
Thank you @dkayiwa. I had initially removed the method from the DAO altogether, but was asked to re-introduce it and deprecate it: https://github.com/openmrs/openmrs-core/pull/3831#discussion_r681121820.

Reply:
Sorry for making you run back and forth. You had done right to remove them.

Reply:
No problem, I've updated the PR :)